_pytest/monkeypatch.py

```diff
@@ -30,8 +30,8 @@ def monkeypatch(request):
     will be raised if the set/deletion operation has no target.
     """
     mpatch = MonkeyPatch()
-    request.addfinalizer(mpatch.undo)
-    return mpatch
+    yield mpatch
+    mpatch.undo
```

Comment:
This should be `.undo()` as we actually want to call the method now, not only pass a callable.